### PR TITLE
Fix ccpcheck

### PR DESCRIPTION
**Pull request by raconte**
Tested using the following command
```
cppcheck --quiet --enable=warning,performance,portability  --inline-suppr -Isrc/lib/ -Isrc/app -DLM_ON  src/
```

```
[src/lib/rest/ConnectionInfo.h:55]: (warning) Member variable 'ConnectionInfo::requestType' is not initialized in the constructor.
[src/lib/rest/ConnectionInfo.h:77]: (warning) Member variable 'ConnectionInfo::requestType' is not initialized in the constructor.
[src/lib/rest/ConnectionInfo.h:99]: (warning) Member variable 'ConnectionInfo::requestType' is not initialized in the constructor.
[src/lib/apiTypesV2/Entity.cpp:128]: (warning) %zd in format string (no. 1) requires a signed integer given in the argument list.
[src/lib/apiTypesV2/Entity.cpp:141]: (warning) %zd in format string (no. 1) requires a signed integer given in the argument list.
[src/lib/cache/subCache.cpp:74]: (performance) Variable 'entityId' is assigned in constructor body. Consider performing initialization in initialization list.
[src/lib/cache/subCache.cpp:75]: (performance) Variable 'entityType' is assigned in constructor body. Consider performing initialization in initialization list.
[src/lib/ngsi/ParseData.h:121]: (warning) Member variable 'QueryContextData::pointNo' is not initialized in the constructor.
[src/lib/ngsi/ParseData.h:121]: (warning) Member variable 'QueryContextData::coords' is not initialized in the constructor.
[src/lib/logMsg/logMsg.cpp:2834] -> [src/lib/logMsg/logMsg.cpp:2835]: (performance) Variable 'nada' is reassigned a value before the old one has been used.
[src/lib/logMsg/logMsg.cpp:2835] -> [src/lib/logMsg/logMsg.cpp:2836]: (performance) Variable 'nada' is reassigned a value before the old one has been used.
[src/lib/logMsg/logMsg.cpp:2836] -> [src/lib/logMsg/logMsg.cpp:2837]: (performance) Variable 'nada' is reassigned a value before the old one has been used.
[src/lib/orionTypes/areas.cpp:69]: (warning) Member variable 'Point::valid' is not initialized in the constructor.
[src/lib/parseArgs/baStd.cpp:169]: (portability) scanf without field width limits can crash with huge input data on some versions of libc.
[src/lib/parseArgs/baStd.cpp:184]: (portability) scanf without field width limits can crash with huge input data on some versions of libc.

```

### CODE REVIEW

**Comment by raconte** [comment]
Fixed this , stills another ones :

1- In src/lib/rest/ConnectionInfo.h
```
[src/lib/rest/ConnectionInfo.h:55]: (warning) Member variable 'ConnectionInfo::requestType' is not initialized in the constructor.
[src/lib/rest/ConnectionInfo.h:77]: (warning) Member variable 'ConnectionInfo::requestType' is not initialized in the constructor.
[src/lib/rest/ConnectionInfo.h:99]: (warning) Member variable 'ConnectionInfo::requestType' is not initialized in the constructor.
```
What values ​​can the  requestType variable take in order to initiliaze them?

2.- In src/lib/parseArgs/baStd.cpp
```
[src/lib/parseArgs/baStd.cpp:169]: (portability) scanf without field width limits can crash with huge input data on some versions of libc.
[src/lib/parseArgs/baStd.cpp:184]: (portability) scanf without field width limits can crash with huge input data on some versions of libc.
```
This a weird crash, it means that with "huge data" is about millions of digits. So, in this case :
  float f;
  double f;
  sscanf(string, "%f", &f);
  sscanf(string, "%lf", &f);
A solution could be --> sscanf(string, "%10f", &f);
But If the code haven´t crash yet, could be considered a false positive.

3.- In src/lib/parseArgs/paConfig.cpp
```
[src/lib/parseArgs/paConfig.cpp:953]: (warning) Comparison of a boolean with an integer that is neither 1 nor 0.
```
This one is because if we see in the paConfig.cpp file ;
```
((paClearAt != -1) || (paKeepLines != -1) || (paLastLines != -1))
```
And the definition of this variable in paBuiltin.cpp it is defined like boolean 
```
  { "-lmca",      &paClearAt,      "CLEAR_AT",    PaInt,  PaOpt, -1, PaNL, PaNL, "clear at lines",      PAI_REST   },
  { "-lmkl",      &paKeepLines,    "KEEP_LINES",  PaInt,  PaOpt, -1, PaNL, PaNL, "clear 'keep lines'",  PAI_REST   },
  { "-lmll",      &paLastLines,    "LAST_LINES",  PaInt,  PaOpt, -1, PaNL, PaNL, "clear 'last lines'",  PAI_REST   },
```
4.- In src/lib/logMsg/logMsg.cp
```
[src/lib/logMsg/logMsg.cpp:2834] -> [src/lib/logMsg/logMsg.cpp:2835]: (performance) Variable 'nada' is reassigned a value before the old one has been used.
[src/lib/logMsg/logMsg.cpp:2835] -> [src/lib/logMsg/logMsg.cpp:2836]: (performance) Variable 'nada' is reassigned a value before the old one has been used.
[src/lib/logMsg/logMsg.cpp:2836] -> [src/lib/logMsg/logMsg.cpp:2837]: (performance) Variable 'nada' is reassigned a value before the old one has been used.
```

File: src/lib/cache/subCache.cpp [hunk 1]
```diff
@@ -71,8 +71,8 @@ volatile SubCacheState subCacheState = ScsIdle;
 */
 EntityInfo::EntityInfo(const std::string& _entityId, const std::string& _entityType, const std::string& _isPattern)
 {
-  entityId     = _entityId;
-  entityType   = _entityType;
+  const std::string entityId     = _entityId;
+  const std::string entityType   = _entityType;
```
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: This looks a little dangerous ...

    entityId     = _entityId;

sets the member 'entityId' to have the value of '_entityId'.

On the other hand:

    const std::string entityId     = _entityId;

just created a local variable and then dies ...

Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: What exactly did cppcheck complain about here?
We'll need a new way to get rid of the warnings ...

Comment (raconte) [inline src/lib/cache/subCache.cpp]:
```
[src/lib/cache/subCache.cpp:74]: (performance) Variable 'entityId' is assigned in constructor body. Consider performing initialization in initialization list.
[src/lib/cache/subCache.cpp:75]: (performance) Variable 'entityType' is assigned in constructor body. Consider performing initialization in initialization list.
```

Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Strange things happening with the comments, perhaps I added a comment in the commit page ...?

You will need to assign  entityId and entityType before the body:

    EntityInfo::EntityInfo(const std::string& _entityId, const std::string& _entityType, const std::string& _isPattern) :
         entityId(_entityId), entityType(_entityType)

Comment (raconte) [inline src/lib/cache/subCache.cpp]: Ok, done 
Fixed in 766b208

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**